benchmarks/nested-objects.bench.ts

```diff
@@ -0,0 +1,74 @@
+import { bench, describe, expect } from 'vitest';
+import { P, match } from '../src';
+
+
+describe('ts-pattern-benchmark/nested-objects', () => {
+  const inputIndex = Math.floor(Math.random() * 3) as 0 | 1 | 2;
+  const input = (() => {
+    const map = {
+      0: { type: 'a' as const, value: { x: Math.random(), y: Math.random() } },
+      1: {
+        type: 'b' as const,
+        value: Math.random() > 0.5 ? [1, 2, 3, 4] : ['hello'],
+      },
+      2: { type: 'c' as const, age: Math.random(), name: 'acdfl' },
+    };
+  
+    return map[inputIndex];
+  })();
```

Comment:
We should re-generate the input on every run instead of only once, otherwise we are always testing against the same object which could make results vary between runs, and could result in irrelevant JIT optimizations.

Reply:
Nevertheless, the object creation time should be excluded from the benchmark.

Therefore, I am looking for ways to use setup.

I will ask you to review it again after working on it.

Reply:
I don't mind that the object creation is included in the benchmark as long as it's included in all cases. Its only purpose is to compare equivalent implementation of the same thing, so the absolute values aren't that important

Reply:
Even if it's an absolute value, it seems to affect the benchmark results.
It doesn't affect who's faster or who's slower
There's definitely an impact when it comes to showing how much faster it is, how much slower it is.

Of course, even if the impact is minimal due to the complexity of the benchmark content.

Is there any problem with JIT about randomly importing pre-generated objects for each benchmark?
i want to import pre-generated objects using random values.